src/frontend/js/components/Icon/index.tsx

```diff
@@ -0,0 +1,58 @@
+interface IconProps {
+  name: IconType;
+  /**
+   * Optional title, will be announced by screen readers
+   * and set as `title` attribute for mouse users
+   *
+   * If omitted, the icon is considered decorative and
+   * ignored by screen readers
+   */
+  title?: string;
+  className?: string;
+}
+
+export const Icon = ({ name, title, className = '' }: IconProps) => {
+  return (
+    <svg
+      {...{
+        className: `icon ${className}`,
+        ...(title
+          ? {
+              role: 'img',
+              'aria-label': title,
+              title,
+            }
+          : {
+              'aria-hidden': true,
+            }),
+      }}
+    >
+      <use href={`#${name}`} />
+    </svg>
+  );
+};
+
+// icons from src/richie/apps/core/templates/richie/icons.html
+type IconType =
```

Comment:
I think it is a good idea as it ease the use of icon during development because before we have to open the `icons.html` file to see the list of available icons.

I'll just move this type above the component declaration.